src/core/ext/filters/client_channel/lb_policy/round_robin/round_robin.c

```diff
@@ -233,6 +234,7 @@ static void rr_subchannel_list_shutdown_and_unref(
       grpc_subchannel_notify_on_state_change(exec_ctx, sd->subchannel, NULL,
                                              NULL,
                                              &sd->connectivity_changed_closure);
+      grpc_subchannel_index_ref();
```

Comment:
This problem affects all LB policies, not just the RR policy.  We also need to change the pick_first and grpclb policies.